app/controllers/amr_heatmap_controller.rb

```diff
@@ -0,0 +1,39 @@
+class AmrHeatmapController < ApplicationController
+  include ReportHelper
+
+  before_action :admin_required
+
+  power :viewable_samples
+
+  # GET /amr_heatmap.json
+  # Return JSON information required to create a visualization,
+  # from submitted parameters
+  def amr_counts
+    samples = []
+    amr_data = []
+
+    params[:id].each do |param_id|
+      sample = current_power.viewable_samples.find(param_id)
+      assert_access
+      samples << sample
+    end
```

Comment:
You should be able to do `current_power.viewable_samples.where(id: params[:sampleIds])` to fetch all the samples in one query.

Reply:
Done!